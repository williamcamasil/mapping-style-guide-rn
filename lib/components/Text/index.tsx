import React from 'react';
import {
  StyleProp, StyleSheet, Text as RNText, TextProps as RNTextProp, TextStyle,
} from 'react-native';

import { useTextStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import {
  ColorPaletesType, FontWeightType, LineHeightType, Typography,
} from '../../tokens';
import {
  getColorStyle, getVariantStyle, getWeightStyle, TextVariantType,
} from './utils';

export type { TextVariantType } from './utils';

export type TextPropTypes = RNTextProp & {
  theme: AppThemeType;
  style?: StyleProp<TextStyle>;
  color?: ColorPaletesType;
  variant?: TextVariantType;
  weight?: FontWeightType;
  lineHeight?: LineHeightType;
};

const styles = StyleSheet.create({
  text: {
    letterSpacing: Typography.letterSpaces.default,
    flexShrink: 1,
  },
});

const Text: React.FC<TextPropTypes> = ({
  children,
  color,
  theme,
  style,
  variant = 'body',
  weight,
  lineHeight,
  allowFontScaling = false,
  ...others
}) => {

  const textStyles = useTextStyles(() => [
    styles.text,
    getColorStyle(theme, color ?? 'neutralGray600'),
    getVariantStyle(theme, variant, lineHeight),
    getWeightStyle(theme, weight),
    style,
  ], [color, lineHeight, style, theme, variant, weight]);

  return (
    <RNText
      {...others}
      allowFontScaling={allowFontScaling}
      style={textStyles}
    >
      {children}
    </RNText>
  );
};

export default withTheme(Text);
