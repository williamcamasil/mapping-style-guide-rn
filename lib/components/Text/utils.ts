import { TextStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { ColorPaletesType, FontWeightType, LineHeightType } from '../../tokens';
import { calcLineHeight } from '../../utils/typography';

export type TextVariantType =
  // heading
  'headingDisplay' |
  'headingXLarge' |
  'headingLarge' |
  'headingMedium' |
  'headingSmall' |
  // paragraph regular
  'subtitle' |
  'body' |
  'bodySmall' |
  'caption' |
  // small
  'small' |
  'smallCaps' |
  // one line
  'oneLineHuge' |
  'oneLineLarge' |
  'oneLineMedium' |
  'oneLineSmall' |
  'oneLineCaption' |
  'oneLineSemibold' |
  'oneLineBoldHuge' |
  'oneLineBoldExtraLarge' |
  'oneLineBoldLarge' |
  'oneLineBoldMedium' |
  'oneLineBoldSmall';

const getHeadingStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  headingDisplay: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.huge,
    lineHeight: calcLineHeight(theme.typography.sizes.huge, lineHeight ?? theme.typography.lineHeights.medium),
  },
  headingXLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XXXL,
    lineHeight: calcLineHeight(theme.typography.sizes.XXXL, lineHeight ?? theme.typography.lineHeights.medium),
  },
  headingLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XXL,
    lineHeight: calcLineHeight(theme.typography.sizes.XXL, lineHeight ?? theme.typography.lineHeights.large),
  },
  headingMedium: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XL,
    lineHeight: calcLineHeight(theme.typography.sizes.XL, lineHeight ?? theme.typography.lineHeights.medium),
  },
  headingSmall: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.large,
    lineHeight: calcLineHeight(theme.typography.sizes.large, lineHeight ?? theme.typography.lineHeights.medium),
  },
});

const getSmallStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  small: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(14, lineHeight ?? theme.typography.lineHeights.medium),
  },
  smallCaps: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XXXS,
    lineHeight: calcLineHeight(theme.typography.sizes.XXXS, lineHeight ?? theme.typography.lineHeights.medium),
    textTransform: 'uppercase' as TextStyle['textTransform'],
  },
});

const getOneLineStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  oneLineHuge: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XL,
    lineHeight: calcLineHeight(theme.typography.sizes.XL, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineLarge: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.large,
    lineHeight: calcLineHeight(theme.typography.sizes.large, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineMedium: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.small,
    lineHeight: calcLineHeight(theme.typography.sizes.small, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineSmall: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(theme.typography.sizes.XS, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineCaption: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XXS,
    lineHeight: calcLineHeight(theme.typography.sizes.XXS, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineSemibold: {
    ...theme.typography.weights.semiBold,
    fontSize: theme.typography.sizes.medium,
    lineHeight: calcLineHeight(theme.typography.sizes.medium, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldHuge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XL,
    lineHeight: calcLineHeight(theme.typography.sizes.XL, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldExtraLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.large,
    lineHeight: calcLineHeight(theme.typography.sizes.large, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldLarge: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.medium,
    lineHeight: calcLineHeight(theme.typography.sizes.medium, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldMedium: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.small,
    lineHeight: calcLineHeight(theme.typography.sizes.small, lineHeight ?? theme.typography.lineHeights.small),
  },
  oneLineBoldSmall: {
    ...theme.typography.weights.bold,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(theme.typography.sizes.XS, lineHeight ?? theme.typography.lineHeights.small),
  },
});

const getParagraphStyles = (theme: AppThemeType, lineHeight: number | null) => ({
  subtitle: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.medium,
    lineHeight: calcLineHeight(theme.typography.sizes.medium, lineHeight ?? theme.typography.lineHeights.large),
  },
  bodySmall: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XS,
    lineHeight: calcLineHeight(theme.typography.sizes.XS, lineHeight ?? theme.typography.lineHeights.large),
  },
  caption: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.XXS,
    lineHeight: calcLineHeight(theme.typography.sizes.XXS, lineHeight ?? theme.typography.lineHeights.medium),
  },
  body: {
    ...theme.typography.weights.regular,
    fontSize: theme.typography.sizes.small,
    lineHeight: calcLineHeight(theme.typography.sizes.small, lineHeight ?? theme.typography.lineHeights.large),
  },
});

export function getVariantStyle(
  theme: AppThemeType,
  variant: TextVariantType,
  lineHeightName?: LineHeightType,
): TextStyle {
  const lineHeight = lineHeightName ? theme.typography.lineHeights[lineHeightName] : null;

  const variantStyles: Record<TextVariantType, TextStyle> = {
    // heading
    ...getHeadingStyles(theme, lineHeight),
    // small
    ...getSmallStyles(theme, lineHeight),
    // one line
    ...getOneLineStyles(theme, lineHeight),
    // paragraph
    ...getParagraphStyles(theme, lineHeight),
  };

  return variantStyles[variant];
}

export function getColorStyle(theme: AppThemeType, color: ColorPaletesType): TextStyle | null {
  return {
    color: theme.colors[color],
  };
}

export function getWeightStyle(theme: AppThemeType, weight?: FontWeightType): TextStyle | null {
  if (!weight) { return null; }

  return theme.typography.weights[weight];
}
