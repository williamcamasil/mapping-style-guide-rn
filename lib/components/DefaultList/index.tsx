import React, {
  useMemo,
  useState,
  useCallback,
  MutableRefObject,
  createRef,
} from 'react';
import {
  RefreshControl,
  FlatList,
  StyleSheet,
  View,
  FlatListProps,
  StyleProp,
  ViewStyle,
  SectionListProps,
  SectionList,
  Platform,
  DefaultSectionT,
} from 'react-native';

import { $DeepPartial, $Without } from '@callstack/react-theme-provider';
import { useDidMountAndUpdate } from 'mapping-context-rn';

import { useViewStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import Loading from '../Loading';
import LoadingContainer from '../LoadingContainer';
import Text from '../Text';

const { OS } = Platform;

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'transparent',
    flex: 1,
  },
  contentContainerStyleEmpty: {
    flex: 1,
  },
  footerContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 8,
  },
  endListMessage: {
    textAlign: 'center',
  },
});

type DefaultListPropsType<Item = any, Section = DefaultSectionT> = Omit<FlatListProps<Item> & SectionListProps<Item, Section>, 'data' | 'sections' | 'type'> & {
  theme: AppThemeType;
  data?: FlatListProps<Item>['data'];
  sections?: SectionListProps<Item, Section>['sections'];
  containerStyle?: StyleProp<ViewStyle>;
  endListMessage?: string;
  loadingMore?: boolean;
  onRefresh?: () => void;
  variant?: 'flat' | 'section';
  innerRef?: MutableRefObject<DefaultListRefType<Item, Section>>;
};

export type DefaultListRefType<Item, Section> = FlatList<Item> & SectionList<Item, Section>;

function DefaultList<Item = any, Section = DefaultSectionT>({
  variant,
  onRefresh,
  refreshing,
  data,
  sections,
  containerStyle,
  contentContainerStyle,
  endListMessage,
  ListEmptyComponent,
  ListFooterComponent,
  loadingMore,
  inverted,
  theme,
  innerRef,
  ...others
}: DefaultListPropsType<Item, Section>): React.ReactElement {

  const [pullToRefresh, setPullToRefresh] = useState(false);

  const ref = useMemo(() => innerRef ?? createRef<DefaultListRefType<Item, Section>>(), [innerRef]);

  const isFlatList = variant === 'flat';
  const isSectionList = variant === 'section';

  /* istanbul ignore next */
  useDidMountAndUpdate(() => {
    if (refreshing && isFlatList) {
      if (OS === 'ios' && !inverted) {
        // força exibição do RefreshControl no iOS
        // https://github.com/facebook/react-native/issues/25898#issuecomment-554107989
        ref.current?.scrollToOffset({ offset: -60, animated: false });
      } else {
        ref.current?.scrollToOffset({ offset: 0 });
      }
    }
  }, [refreshing]);

  const dataArray = useMemo(() => (isSectionList ? sections : data), [isSectionList, sections, data]);

  const hasDataArray = Boolean(dataArray?.length);

  const handleRefresh = useCallback(() => {
    setPullToRefresh(true);
    /* istanbul ignore else  */
    if (onRefresh) {
      onRefresh();
    }
  }, [onRefresh]);

  const refreshControlColors = useMemo(() => [theme.colors.primaryMain], [theme.colors.primaryMain]);

  const contentContainerStyles = useViewStyles(
    () => [
      !hasDataArray ? styles.contentContainerStyleEmpty : null,
      contentContainerStyle,
    ],
    [contentContainerStyle, hasDataArray],
  );

  const listFooter = useMemo(() => {
    if (refreshing) return null;

    if (loadingMore) {
      return (
        <View style={styles.footerContainer}>
          <Loading color="secondaryMain" />
        </View>
      );
    }

    if (ListFooterComponent) return ListFooterComponent;

    if (hasDataArray && endListMessage) {
      return (
        <View style={styles.footerContainer}>
          <Text style={styles.endListMessage} color="feedbackError300">
            {endListMessage}
          </Text>
        </View>
      );
    }

    return null;
  }, [loadingMore, ListFooterComponent, endListMessage, hasDataArray, refreshing]);

  const renderList = () => {
    const emptyContent = refreshing ? null : ListEmptyComponent;

    const refreshComponent = onRefresh ? (
      <RefreshControl
        testID="default-list-refresh-control"
        onRefresh={handleRefresh}
        refreshing={Boolean(refreshing && (hasDataArray || pullToRefresh))}
        colors={refreshControlColors}
        tintColor={theme.colors.primaryMain}
      />
    ) : undefined;

    if (isSectionList && sections) {
      return (
        <SectionList<Item, Section>
          testID="internal-list"
          scrollToOverflowEnabled
          {...others}
          sections={sections}
          refreshControl={refreshComponent}
          contentContainerStyle={contentContainerStyles}
          ListEmptyComponent={emptyContent}
          ListFooterComponent={listFooter}
          stickySectionHeadersEnabled={false}
          inverted={inverted}
          ref={ref}
        />
      );
    }

    return (
      <FlatList<Item>
        testID="internal-list"
        scrollToOverflowEnabled
        {...others}
        data={data}
        refreshControl={refreshComponent}
        contentContainerStyle={contentContainerStyles}
        ListEmptyComponent={emptyContent}
        ListFooterComponent={listFooter}
        inverted={inverted}
        ref={ref}
      />
    );
  };

  const renderLoadingContainer = () => {
    if (!hasDataArray && refreshing) {
      return (
        <LoadingContainer />
      );
    }
    return null;
  };

  const containerStyles = useViewStyles(() => [
    styles.container,
    containerStyle,
  ], [containerStyle]);

  return (
    <View style={containerStyles}>
      {renderList()}
      {renderLoadingContainer()}
    </View>
  );
}

DefaultList.defaultProps = {
  type: 'flat',
};

/*
 * Devido ao componente trabalhar com "generics" e tbm necessitar de um funções de
 * alta ordem (wrapModal, withTheme, wrapField, etc),
 * é necessário forçar a tipagem de saída devido a uma limitação do TypeScript.
 * https://stackoverflow.com/q/58469229/2826279
 */
// @ts-ignore
export default withTheme(DefaultList) as <Item = any, Section = DefaultSectionT>(
  props: $Without<DefaultListPropsType<Item, Section>, 'theme'> & { theme?: $DeepPartial<AppThemeType> },
) => React.ReactElement;
