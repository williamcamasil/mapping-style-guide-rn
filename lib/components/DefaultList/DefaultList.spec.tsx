import React from 'react';
import { RefreshControl, SectionListData } from 'react-native';

import { render, fireEvent } from '@testing-library/react-native';

import DefaultList from '.';
import Text from '../Text';

type PersonType = {
  name: string;
  age: number;
};

type PersonSectionType = SectionListData<PersonType, { title: string }>;

const PERSONS_LIST: PersonType[] = [
  {
    name: 'Douglas',
    age: 33,
  },
  {
    name: 'Bruna',
    age: 34,
  },
  {
    name: 'Arthur',
    age: 2,
  },
];

const PERSONS_SECTIONS: PersonSectionType[] = [
  {
    title: 'Homens',
    data: [
      {
        name: 'Douglas',
        age: 33,
      },

      {
        name: 'Arthur',
        age: 2,
      },
    ],
  },
  {
    title: 'Mulheres',
    data: [
      {
        name: 'Bruna',
        age: 34,
      },
    ],
  },
];

describe('DefaultList snapshot', () => {
  it('flat', () => {
    const tree = render(
      <DefaultList<PersonType>
        data={PERSONS_LIST}
        renderItem={({ item }) => (
          <Text>
            {item.name}
            {' - '}
            {item.age}
          </Text>
        )}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('sections', () => {
    const tree = render(
      <DefaultList<PersonType, PersonSectionType>
        variant="section"
        sections={PERSONS_SECTIONS}
        renderSectionHeader={({ section }) => (
          <Text>{section.title}</Text>
        )}
        renderItem={({ item }) => (
          <Text>
            {item.name}
            {' - '}
            {item.age}
          </Text>
        )}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('ListFooterComponent', () => {
    const tree = render(
      <DefaultList<PersonType, PersonSectionType>
        data={PERSONS_LIST}
        renderItem={({ item }) => (
          <Text>
            {item.name}
            {' - '}
            {item.age}
          </Text>
        )}
        ListFooterComponent={(
          <Text>Footer</Text>
        )}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('endListMessage', () => {
    const tree = render(
      <DefaultList<PersonType, PersonSectionType>
        data={PERSONS_LIST}
        renderItem={({ item }) => (
          <Text>
            {item.name}
            {' - '}
            {item.age}
          </Text>
        )}
        endListMessage="Fim da lista"
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('DefaultList events', () => {
  const handleRefresh = jest.fn();

  const tree = render(
    <DefaultList<PersonType, PersonSectionType>
      sections={PERSONS_SECTIONS}
      renderItem={({ item }) => (
        <Text>
          {item.name}
          {' - '}
          {item.age}
        </Text>
      )}
      onRefresh={handleRefresh}
    />,
  );

  const flatList = tree.UNSAFE_getByType(RefreshControl);
  fireEvent(flatList, 'refresh');

  expect(handleRefresh).toBeCalled();
});
