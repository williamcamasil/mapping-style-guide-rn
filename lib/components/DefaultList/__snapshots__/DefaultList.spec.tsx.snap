// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultList snapshot ListFooterComponent 1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListFooterComponent={
      <withTheme(Text)>
        Footer
      </withTheme(Text)>
    }
    contentContainerStyle={{}}
    data={
      [
        {
          "age": 33,
          "name": "Douglas",
        },
        {
          "age": 34,
          "name": "Bruna",
        },
        {
          "age": 2,
          "name": "Arthur",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    scrollToOverflowEnabled={true}
    stickyHeaderIndices={[]}
    testID="internal-list"
    type="flat"
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Douglas
           - 
          33
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Bruna
           - 
          34
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Arthur
           - 
          2
        </Text>
      </View>
      <View
        onLayout={[Function]}
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Footer
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`DefaultList snapshot endListMessage 1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListFooterComponent={
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <withTheme(Text)
          color="feedbackError300"
          style={
            {
              "textAlign": "center",
            }
          }
        >
          Fim da lista
        </withTheme(Text)>
      </View>
    }
    contentContainerStyle={{}}
    data={
      [
        {
          "age": 33,
          "name": "Douglas",
        },
        {
          "age": 34,
          "name": "Bruna",
        },
        {
          "age": 2,
          "name": "Arthur",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    scrollToOverflowEnabled={true}
    stickyHeaderIndices={[]}
    testID="internal-list"
    type="flat"
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Douglas
           - 
          33
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Bruna
           - 
          34
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Arthur
           - 
          2
        </Text>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              {
                "color": "#FF8E88",
                "flexShrink": 1,
                "fontFamily": "Averta CY",
                "fontSize": 16,
                "fontWeight": "normal",
                "letterSpacing": -0.02,
                "lineHeight": 22,
                "textAlign": "center",
              }
            }
          >
            Fim da lista
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`DefaultList snapshot flat 1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListFooterComponent={null}
    contentContainerStyle={{}}
    data={
      [
        {
          "age": 33,
          "name": "Douglas",
        },
        {
          "age": 34,
          "name": "Bruna",
        },
        {
          "age": 2,
          "name": "Arthur",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    scrollToOverflowEnabled={true}
    stickyHeaderIndices={[]}
    testID="internal-list"
    type="flat"
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Douglas
           - 
          33
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Bruna
           - 
          34
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Arthur
           - 
          2
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`DefaultList snapshot sections 1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListFooterComponent={null}
    contentContainerStyle={{}}
    data={
      [
        {
          "data": [
            {
              "age": 33,
              "name": "Douglas",
            },
            {
              "age": 2,
              "name": "Arthur",
            },
          ],
          "title": "Homens",
        },
        {
          "data": [
            {
              "age": 34,
              "name": "Bruna",
            },
          ],
          "title": "Mulheres",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    scrollToOverflowEnabled={true}
    stickyHeaderIndices={[]}
    testID="internal-list"
    type="flat"
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Homens
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Douglas
           - 
          33
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Arthur
           - 
          2
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      />
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Mulheres
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          style={
            {
              "color": "#676D79",
              "flexShrink": 1,
              "fontFamily": "Averta CY",
              "fontSize": 16,
              "fontWeight": "normal",
              "letterSpacing": -0.02,
              "lineHeight": 22,
            }
          }
        >
          Bruna
           - 
          34
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
</View>
`;
